pkg/services/updatechecker/plugins.go

```diff
@@ -215,3 +242,30 @@ func (s *PluginsService) pluginsEligibleForVersionCheck(ctx context.Context) map
 
 	return result
 }
+func (s *PluginsService) updateAll(ctx context.Context) {
+	ctxLogger := s.log.FromContext(ctx)
+
+	failedUpdates := map[string]string{}
+
+	for pluginID, updateVersion := range s.availableUpdates {
+		compatOpts := plugins.NewAddOpts(s.grafanaVersion, runtime.GOOS, runtime.GOARCH, "")
+
+		plugin, exists := s.pluginStore.Plugin(ctx, pluginID)
+		if !exists {
+			ctxLogger.Error("Plugin not found", "pluginID", pluginID)
+			continue
+		}
```

Comment:
rather than requesting the plugin again, the current plugin version is already available when generating `s.availableUpdates` so I'd change the type of map from `map[string]string` to a `map[string]struct{localVersion, availableVersion}`